### PR TITLE
rustc: don't trip an assertion for enums with present but uninhabited variants.

### LINKED ISSUE #50731: wasm build error in nightly-2018-05-13
Looks like something broke building wasm in this range: https://github.com/rust-lang/rust/compare/ff2ac35db...9fae15374

This is the error:
```
thread 'main' panicked at 'discriminant range is 170141183460469231731687303715884105727...-170141183460469231731687303715884105728', librustc/ty/layout.rs:891:17
note: Run with `RUST_BACKTRACE=1` for a backtrace.
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: rustc 1.27.0-nightly (9fae15374 2018-05-13) running on x86_64-unknown-linux-gnu
note: compiler flags: -C opt-level=3 -C panic=abort --crate-type lib
note: some of the compiler flags provided by cargo are hidden
```

https://travis-ci.org/LiveSplit/livesplit-core/jobs/378520318#L551

### PR DESCRIPTION
**Pull request by eddyb**
Fixes #50731.

r? @nikomatsakis

### CODE REVIEW

**Review by nikomatsakis — APPROVED** [review]

**Comment by nikomatsakis** [comment]
@bors r+

cc @rust-lang/compiler -- approving for backport

File: src/librustc/ty/layout.rs [hunk 1]
```diff
@@ -888,6 +888,11 @@ impl<'a, 'tcx> LayoutCx<'tcx, TyCtxt<'a, 'tcx, 'tcx>> {
                     if x < min { min = x; }
                     if x > max { max = x; }
                 }
+                // We might have no inhabited variants, so pretend there's at least one.
+                if (min, max) == (i128::max_value(), i128::min_value()) {
```
Comment (estebank) [inline src/librustc/ty/layout.rs]: Just to make sure, is `(min, max) = max(), min()` correct?

Comment (oli-obk) [reply src/librustc/ty/layout.rs]: Jup. These are the initial values for both, so that means the loop was empty or all variants uninhabited

**Pull request merged**